server/auth/me.js

```diff
@@ -2,61 +2,79 @@ const router = require('express').Router();
 const db = require('../db')
 const User = db.models.user;
 
-router.route('/') 
-  .get((req, res, next) => {              //get user
-      if (req.user) {
-          res.json(req.user.sanitize())
-      } else {
-          res.json({})
-      }
-  })
-  .post((req, res, next) => {           // create user
-    User.findOrCreate({
-        where: {
-            email: req.body.email
-        },
-        defaults: {
-            password: req.body.password,
-            username: req.body.username,
-            points: 0
-        }
-    })
-    .spread((user, exist) => {
-      if (exist) req.login(user, error => {
-        if (error) {
-            next(error)
+router.route('/')
+    .get((req, res, next) => {              //get user
+        if (req.user) {
+            res.json(req.user.sanitize())
         } else {
-            let u = user.sanitize()
-            res.json(u)
+            res.json({})
         }
-        });
-        if (!exist) res.status(401).send('User exists. Please log in.')})
-        .catch(next)
+    })
+    .post((req, res, next) => {           // create user
+        User.findOrCreate({
+            where: {
+                email: req.body.email
+            },
+            defaults: {
+                password: req.body.password,
+                username: req.body.username,
+                points: 0
+            }
+        })
+            .spread((user, notExist) => {
+                if (notExist) req.login(user, error => {
+                    if (error) {
+                        next(error)
+                    } else {
+                        let u = user.sanitize()
+                        res.json(u)
+                    }
+                });
+                if (!notExist) res.json({signupError: 'Failed to create account. User might already exist.'})
```

Comment:
does this need to be an if / if statement or can we make this into a if / else if or if / else it seems pretty  TRUE or FALSE here  @AnnaC91